packages/paste-core/components/table/src/types.ts

```diff
@@ -0,0 +1,76 @@
+import * as PropTypes from 'prop-types';
+import {ValueOf} from '@twilio-paste/types';
+import {WidthOptions, isWidthTokenProp} from '@twilio-paste/style-props';
+import {TableAlignmentObject, TableLayoutObject, TableVerticalAlignmentObject} from './constants';
+
+export type TableAlignmentOptions = ValueOf<typeof TableAlignmentObject>;
+export type TableLayoutOptions = ValueOf<typeof TableLayoutObject>;
+export type TableVerticalAlignmentOptions = ValueOf<typeof TableVerticalAlignmentObject>;
+
+export interface TableProps {
+  children: NonNullable<React.ReactNode>;
+  tableLayout?: TableLayoutOptions;
+}
+
+export const TablePropTypes = {
+  children: PropTypes.node.isRequired,
+  tableLayout: PropTypes.oneOf(Object.keys(TableLayoutObject).map(tableLayout => TableLayoutObject[tableLayout])),
+};
+
+export interface THeadProps {
+  children: NonNullable<React.ReactNode>;
+}
+
+export const THeadPropTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+export interface TBodyProps {
+  children: NonNullable<React.ReactNode>;
+}
+
+export const TBodyPropTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+export interface TrProps {
+  children: NonNullable<React.ReactNode>;
+  verticalAlign?: TableVerticalAlignmentOptions;
+}
+
+export const TrPropTypes = {
+  children: PropTypes.node.isRequired,
+  verticalAlign: PropTypes.oneOf(
+    Object.keys(TableVerticalAlignmentObject).map(verticalAlignment => TableVerticalAlignmentObject[verticalAlignment])
```

Comment:
can't you do `Object.values(TableVerticalAlignmentObject)` for these?

Reply:
Much simpler.